keps/sig-network/2157-kube-proxy-config/README.md

```diff
@@ -0,0 +1,207 @@
+- [ ] - [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] - [ ] (R) Design details are appropriately documented
+- [ ] - [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] - [ ] (R) Graduation criteria is in place
+- [ ] - [ ] (R) Production readiness review completed
+- [ ] - [ ] Production readiness review approved
+- [ ] - [ ] "Implementation History" section is up-to-date for milestone
+- [ ] - [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] - [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+The current kube-proxy flags are largely ignored by default, but they **are** validated. This is self-consistent
+but misleading. As an example, you can't send `--blah`, but you can say `--proxy-mode=thisiswrong`.
+The latter is essentially a no-op (it should break the kube-proxy and "fail closed" as Tim might say :).
+
+We thus propose a few simple consistency amendments to the input handling of the kube-proxy which will
+be self-consistent and prevent "no-op" inputs that are silently ignored, and in general, suggest transparency
+in terms of what inputs to kube-proxy are being utilized. The KEP intends to enhance the visibility of the flags already deprecated and provides consistency between flags and
+the existent component configuration API in kube-proxy as stated on [KEP 783](https://github.com/kubernetes/enhancements/tree/master/keps/sig-cluster-lifecycle/wgs/783-component-base).
+
+## Motivation
+
+The latter command ALSO ignores correct flags `--proxy-mode=iptables`, as you might expect,
+meaning that users might try to configure something and then spend several minutes trying to figure out why
+their config wasn't exercised. Since there are several kube-proxy options that might be overridden (there are ~50 options configured in a typical
+config-map for the kube proxy), this is obviously a compounding problem.
+
+The lack of flag precedence over the configuration API, option defaulting, and validation with proper round trip
+test contributes to increasing the existent or newly added flags' inconsistency. These statements are a requirement for the
+future configuration API graduation covered on [KEP 784](https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/784-kube-proxy-component-config).
+
+## Goals
+
+We thus suggest the following modifications to kube-proxy input handling.
+
+1) For some options (such as the `mode`) the configuration is printed.  We suggest that all options be printed after
```

Comment:
If we use -v=5 all flags are printed by default, after finishing option 3), printing the configuration struct (which has the flags merged) should be enough.